editor/docks/filesystem_dock.cpp

```diff
@@ -3566,8 +3575,16 @@ void FileSystemDock::_tree_empty_click(const Vector2 &p_pos, MouseButton p_butto
 	current_path = "res://";
 	tree_popup->clear();
 	tree_popup->reset_size();
-
-	_set_popup_create_options(tree_popup);
+	tree_popup->add_icon_item(get_editor_theme_icon(SNAME("Folder")), TTRC("New Folder..."), FILE_MENU_NEW_FOLDER);
+	tree_popup->add_icon_item(get_editor_theme_icon(SNAME("PackedScene")), TTRC("New Scene..."), FILE_MENU_NEW_SCENE);
+	tree_popup->add_icon_item(get_editor_theme_icon(SNAME("Script")), TTRC("New Script..."), FILE_MENU_NEW_SCRIPT);
+	tree_popup->add_icon_item(get_editor_theme_icon(SNAME("Object")), TTRC("New Resource..."), FILE_MENU_NEW_RESOURCE);
+	tree_popup->add_icon_item(get_editor_theme_icon(SNAME("TextFile")), TTRC("New TextFile..."), FILE_MENU_NEW_TEXTFILE);
+	tree_popup->set_item_shortcut(tree_popup->get_item_index(FILE_MENU_NEW_FOLDER), ED_GET_SHORTCUT("filesystem_dock/new_folder"));
+	tree_popup->set_item_shortcut(tree_popup->get_item_index(FILE_MENU_NEW_SCENE), ED_GET_SHORTCUT("filesystem_dock/new_scene"));
+	tree_popup->set_item_shortcut(tree_popup->get_item_index(FILE_MENU_NEW_SCRIPT), ED_GET_SHORTCUT("filesystem_dock/new_script"));
+	tree_popup->set_item_shortcut(tree_popup->get_item_index(FILE_MENU_NEW_RESOURCE), ED_GET_SHORTCUT("filesystem_dock/new_resource"));
+	tree_popup->set_item_shortcut(tree_popup->get_item_index(FILE_MENU_NEW_TEXTFILE), ED_GET_SHORTCUT("filesystem_dock/new_textfile"));
```

Comment:
Instead of using `get_item_index()`, you could assign shortcut right after adding item, and use `-1` as index (which means last added item).